### PR TITLE
New Application States

### PR DESCRIPTION
**Pull request by EasySouls**
Added the new application states in both the backend and frontend, and migrated the database.
Also created a seed file to ease the development.

There are 3 main new functionalities:
- A switch which enables automatic status change upon exporting passes and applications
- A button which sets all application which status Distributed to WaitingForOps
- If the auto status change is enabled, the onPassExport and onApplicationExport functions change the state to the next one for every selected application

I couldn't really find a good place for the auto status change switch, so it looks kind of out of place, I am open to suggestions.
I only tested the new functionalities manually as we do not have tests with 20 or so test data, but it worked for me.

### CODE REVIEW

**Comment by EasySouls** [comment]
Now whether the statuses should automatically be changed show up every time before exporting applications and passes as a dialog, and not a switch above the data. I left the switch code in as well commented out, if the body circle member would like to just set the option once via the switch and not be asked every time in the future.

File: apps/frontend/src/app/periods/[id]/columns.tsx [hunk 1]
```diff
@@ -269,7 +269,7 @@ export const columns: (
               variant='outline'
               className='h-min px-2 py-0.5 rounded'
               disabled={(row.original.status as ApplicationStatus) === ('REJECTED' as ApplicationStatus)}
-              onClick={() => onStatusChange(row.original, ApplicationStatus.FINISHED)}
+              onClick={() => onStatusChange(row.original, ApplicationStatus.VALID)}
```
Comment (mozsarmate) [inline apps/frontend/src/app/periods/[id]/columns.tsx:272]: DISTRIBUTED inkább, nem?

File: apps/frontend/src/app/periods/[id]/data-table.tsx [hunk 2]
```diff
@@ -226,12 +230,27 @@ export function DataTable<TData, TValue>({
...
+        <div className='flex gap-2'>
+          <TooltipProvider>
+            <Tooltip>
+              <TooltipTrigger asChild>
+                <Button onClick={() => onSetToDistributedClicked(data)} variant='outline'>
+                  Nyomtatással kész
+                </Button>
+              </TooltipTrigger>
+              <TooltipContent>
+                <p>Az összes NYOMTATÁSRA KÉSZ státusszal rendelkező applikációt átállítja KIOSZTOTT-ra</p>
```
Comment (mozsarmate) [inline apps/frontend/src/app/periods/[id]/data-table.tsx:242]: Tegedheztjük a user-t azért
de am ez a Tooltip szerintem elég patika

File: apps/frontend/src/app/periods/[id]/page.tsx [hunk 3]
```diff
@@ -92,6 +190,17 @@ export default function Page({ params }: { params: { id: number } }) {
   return (
     <div className={quickModeEnabled ? '2xl:-mx-64 xl:-mx-32 max-xl:-mx-8 max-md:-mx-4 px-4 py-0 -mt-4' : ''}>
       <GeneratingDialog open={generatingDialogOpened} />
+      <AcceptDialog
+        open={autoChangeStatusDialogOpened}
+        title='Automatikus státuszváltás'
+        description='Szeretné, hogy az érintett applikációk státuszai automatikusan változzanak exportáláskor?'
```
Comment (mozsarmate) [inline apps/frontend/src/app/periods/[id]/page.tsx:196]: Itt is lehetne tegezés, de a dialog kircsi

File: apps/frontend/src/components/ui/StatusBadge.tsx [hunk 4]
```diff
@@ -30,15 +50,25 @@ export default function StatusBadge({ status }: { status: ApplicationStatus }) {
...
+      case ApplicationStatus.DISTRIBUTED:
+        return <FiTruck />;
+      case ApplicationStatus.WAITING_FOR_OPS:
+        return <FiPauseCircle />;
+      case ApplicationStatus.VALID:
         return <FiArrowRightCircle />;
+      case ApplicationStatus.REVOKED:
+        return <FiRotateCcw />;
+      case ApplicationStatus.EXPIRED:
+        return <FiClock />;
```
Comment (mozsarmate) [inline apps/frontend/src/components/ui/StatusBadge.tsx:66]: they look nice !!

File: apps/frontend/src/components/ui/accept-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      <DialogContent className='sm:max-w-[425px]' closable={false}>
+        <DialogHeader>
+          <DialogTitle>{title}</DialogTitle>
+          <DialogDescription className='my-8'>{description}</DialogDescription>
+        </DialogHeader>
+        <div className='flex gap-4'>
+          <Button onClick={onAccept} variant='default'>
+            Igen
+          </Button>
+          <Button onClick={onDecline} variant='destructive'>
```
Comment (mozsarmate) [inline apps/frontend/src/components/ui/accept-dialog.tsx:24]: variant=seconday?

Comment (mozsarmate) [reply apps/frontend/src/components/ui/accept-dialog.tsx:24]: secondary

**Review by mozsarmate — COMMENTED** [review]
overall the changes look nice, good job

i have not been able to try them with a local db,
please wait @FearsomeRover 's review (or mine latter one (tomorrow))

File: apps/frontend/src/components/ui/accept-dialog.tsx (hunk 5 above)
Comment (EasySouls) [reply apps/frontend/src/components/ui/accept-dialog.tsx:24]: Az accept vagy a decline buttonra gondoltál?

File: apps/frontend/src/app/periods/[id]/data-table.tsx (hunk 2 above)
Comment (EasySouls) [reply apps/frontend/src/app/periods/[id]/data-table.tsx:242]: Én ezt a gomb működésére gondoltam, hogy a tooltip csak közli, hogy mit fog csinálni, nem az user megszólításaként.

File: apps/frontend/src/app/periods/[id]/columns.tsx (hunk 1 above)
Comment (EasySouls) [reply apps/frontend/src/app/periods/[id]/columns.tsx]: Így jó, hogy ha nem DISTRIBUTED, akkor megjelenítjük a kiosztás gombot, ami átállítja DISTRIBUTED-ra? Vagy csak akkor legyen megjelenítve, ha a DISTRIBUTED előtti státuszban van, ami a MANUFACTURED, és az összes többi esetben el van rejtve?

File: apps/frontend/src/components/ui/accept-dialog.tsx (hunk 5 above)
Comment (EasySouls) [reply apps/frontend/src/components/ui/accept-dialog.tsx]: Közben rájöttem h a decline buttonra értetted a kommentet.

**Review by FearsomeRover — APPROVED** [review]
Nicee

**Pull request merged**